sys/riotboot/flashwrite.c

```diff
@@ -178,10 +173,14 @@ int riotboot_flashwrite_invalidate(int slot)
         return -2;
     }
 
-    uint8_t data_flash[4];
-    memset(data_flash, INVALIDATE_HDR, sizeof(data_flash));
+    /* invalidate header and checksum, write the whole header to avoid
+       running in memory alignment issues with FLASHPAGE_WRITE_BLOCK_SIZE */
+    riotboot_hdr_t tmp_hdr;
+    memcpy(&tmp_hdr, riotboot_slot_get_hdr(slot), sizeof(riotboot_hdr_t));
+    memset(&tmp_hdr.magic_number, (~FLASHPAGE_ERASE_STATE), sizeof(tmp_hdr.magic_number));
+    memset(&tmp_hdr.chksum, (~FLASHPAGE_ERASE_STATE), sizeof(tmp_hdr.chksum));
```

Comment:
```suggestion
    riotboot_hdr_t tmp_hdr = {0};
```

We want to erase the header, so no need to back it up.

Reply:
sure

Reply:
Addressed